test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.Document.Tests/Get_ApplicationJson_DictionaryObject_Tests.cs

```diff
@@ -84,5 +87,29 @@ public void Given_OpenApiDocument_Then_It_Should_Return_ComponentSchemaProperty(
 
             this._doc["components"]["schemas"][itemRef].Should().NotBeNullOrEmpty();
         }
+
+        [DataTestMethod]
+        [DataRow("string[]_string", "array")]
+        [DataRow("int32[]_int32", "array")]
+        public void Given_OpenApiDocument_Then_It_Should_Return_ComponentReferenceSchema(string @ref, string refType)
+        {
+            var schemas = this._doc["components"]["schemas"];
+            var schema = schemas[@ref]["type"];
+
+            schema.Should().NotBeNull();
+            schema.Value<string>().Should().Be(refType);
+            schema.Value<string>().Should().Be(refType);
+        }
+
+        [DataTestMethod]
+        [DataRow("string[]_string", "array", "string")]
+        [DataRow("int32[]_int32", "array", "integer")]
```

Comment:
In the OpenAPI spec, the Dictionary type is always Dictionary<string, ....>. It can't be Dictionary<object, ...>.